pkg/cmd/step/step_next_version_integration_test.go

```diff
@@ -112,3 +112,59 @@ func TestSetVersionChart(t *testing.T) {
 
 	assert.Equal(t, string(testFile), string(updatedFile), "replaced version")
 }
+
+func TestRunChartsDir(t *testing.T) {
+	originalJxHome, tempJxHome, err := testhelpers.CreateTestJxHomeDir()
+	assert.NoError(t, err)
+	defer func() {
+		err := testhelpers.CleanupTestJxHomeDir(originalJxHome, tempJxHome)
+		assert.NoError(t, err)
+	}()
+	originalKubeCfg, tempKubeCfg, err := testhelpers.CreateTestKubeConfigDir()
+	assert.NoError(t, err)
+	defer func() {
+		err := testhelpers.CleanupTestKubeConfigDir(originalKubeCfg, tempKubeCfg)
+		assert.NoError(t, err)
+	}()
+
+	f, err := ioutil.TempDir("", "test-run-chartsdir")
+	assert.NoError(t, err)
+
+	testData := path.Join("test_data", "next_version", "charts")
+	_, err = os.Stat(testData)
+	assert.NoError(t, err)
+
+	err = util.CopyDir(testData, f, true)
+	assert.NoError(t, err)
+
+	git := gits.NewGitCLI()
+	err = git.Init(f)
+	assert.NoError(t, err)
+
+	o := step.StepNextVersionOptions{
+		StepOptions: step2.StepOptions{
+			CommonOptions: &opts.CommonOptions{},
+		},
+	}
+	o.Out = tests.Output()
+	o.Dir = f
+	o.NewVersion = "1.2.3"
+	o.Tag = true
+	o.ChartsDir = path.Join(f, "charts", "foo")
+	o.SetGit(&gits.GitFake{})
+
+	err = o.Run()
+	assert.NoError(t, err)
+
+	os.Remove("VERSION")
```

Comment:
fyi, StepNextVersionOptions.Run has the side-effect of writing out a VERSION file, which ends up in the working directory (so `pkg/cmd/step` in this instance). Any better way to do this? Can we change the working directory for the scope of the test to the temporary directory returned by `ioutil.Tempdir("", "test-run-chartsdir")` ?

Reply:
> fyi, StepNextVersionOptions.Run has the side-effect of writing out a VERSION file, which ends up in the working directory (so pkg/cmd/step in this instance)

:-( 

> Any better way to do this? Can we change the working directory for the scope of the test to the temporary directory returned by ioutil.Tempdir("", "test-run-chartsdir")

> Any better way to do this? Can we change the working directory for the scope of the test to the temporary directory returned by ioutil.Tempdir("", "test-run-chartsdir")

+1, I think this is the best option for now. Also, the cleanup should happen in a `defer` so that it really occurs, even in the case of an unexpected error